include/deal.II/distributed/fully_distributed_tria.h

```diff
@@ -0,0 +1,406 @@
+       * the constructor. Settings can be combined using bitwise OR.
+       */
+      enum Settings
+      {
+        /**
+         * Default settings, other options are disabled.
+         */
+        default_setting = 0x0,
+        /**
+         * This flags needs to be set to use the geometric multigrid
+         * functionality. This option requires additional computation and
+         * communication.
+         */
+        construct_multigrid_hierarchy = 0x1
+      };
+
+      /**
+       * Constructor.
+       *
+       * @param mpi_communicator denotes the MPI communicator to be used for the
+       *                         triangulation.
+       * @param settings See the description of the Settings enumerator.
+       */
+      explicit Triangulation(MPI_Comm       mpi_communicator,
+                             const Settings settings = default_setting);
+
+      /**
+       * Destructor.
+       */
+      virtual ~Triangulation() = default;
+
+      /**
+       * Create a triangulation from the provided ConstructionData.
+       *
+       * @note The namespace dealii::fullydistributed::Util contains functions
+       *       to create ConstructionData.
+       *
+       * @note This is the function to be used instead of
+       * Triangulation::create_triangulation() for some of the other
+       * triangulations of deal.II.
+       *
+       * @param construction_data The data needed for this process.
+       */
+      void
+      reinit(const ConstructionData<dim, spacedim> &construction_data);
+
+      /**
+       * @note This function is not implemented for this class  and throws
+       *       an assertion. Instead, use
+       *       the function reinit() to create the triangulation.
+       */
+      virtual void
+      create_triangulation(const std::vector<Point<spacedim>> &vertices,
+                           const std::vector<CellData<dim>> &  cells,
+                           const SubCellData &subcelldata) override;
+
+      /**
+       * Coarsen and refine the mesh according to refinement and coarsening
+       * flags set.
+       *
+       * @note Not implemented yet.
+       */
+      virtual void
+      execute_coarsening_and_refinement() override;
+
+      /**
+       * Override the implementation of prepare_coarsening_and_refinement from
+       * the base class.
+       *
+       * @note Not implemented yet.
+       */
+      virtual bool
+      prepare_coarsening_and_refinement() override;
+
+      /**
+       * Return true if the triangulation has hanging nodes.
+       *
+       * @note Not implemented yet.
+       */
+      virtual bool
+      has_hanging_nodes() const override;
+
+      /**
+       * Return the local memory consumption in bytes.
+       */
+      virtual std::size_t
+      memory_consumption() const override;
+
+      /**
+       * Override the function to update the number cache so we can fill data
+       * like level_ghost_owners.
```

Comment:
This is a description for those who know the implementation of these classes, but it doesn't provide any useful information to people who only *use* the class. I'd just omit the description, in which case doxygen will copy the description from the base class.

Separately: Shouldn't this function be `private`?

Reply:
- you are right `update_number_cache` should be private
- the comment (copied from `p:d:t`) describes what happens additionally to the base class: so I would keep it...